src/ios/CDVLocation.m

```diff
@@ -80,87 +81,111 @@ - (BOOL)isAuthorized
     return YES;
 }
 
-- (BOOL)isLocationServicesEnabled
-{
-    BOOL locationServicesEnabledInstancePropertyAvailable = [self.locationManager respondsToSelector:@selector(locationServicesEnabled)]; // iOS 3.x
-    BOOL locationServicesEnabledClassPropertyAvailable = [CLLocationManager respondsToSelector:@selector(locationServicesEnabled)]; // iOS 4.x
-
-    if (locationServicesEnabledClassPropertyAvailable) { // iOS 4.x
-        return [CLLocationManager locationServicesEnabled];
-    } else {
-        return NO;
-    }
+- (void)isLocationServicesEnabledWithCompletion:(void (^)(BOOL enabled))completion {
+    dispatch_async(queue, ^{
+        BOOL locationServicesEnabledClassPropertyAvailable = [CLLocationManager respondsToSelector:@selector(locationServicesEnabled)]; // iOS 4.x
+        
+        BOOL result = NO;
+        if (locationServicesEnabledClassPropertyAvailable) { // iOS 4.x
+            result = [CLLocationManager locationServicesEnabled];
+        }
+        
+        dispatch_async(dispatch_get_main_queue(), ^{
```

Comment:
Do we need to call the completion handler on the main queue?

Reply:
Not really, but before it was running on the main thread so I left it the same way.